build/launch_training.py

```diff
@@ -52,66 +60,109 @@ def main():
+                full_checkpoint_dir = os.path.join(
+                    training_args.output_dir, lora_checkpoint_dir
+                )
+
+                logging.info(
+                    "Merging lora tuned checkpoint %s with base model into output path: %s",
+                    lora_checkpoint_dir,
+                    export_path,
+                )
+
+                create_merged_model(
+                    checkpoint_models=full_checkpoint_dir,
+                    export_path=export_path,
+                    base_model=model_args.model_name_or_path,
+                    save_tokenizer=True,
+                )
+            except Exception as e:  # pylint: disable=broad-except
+                logging.error(traceback.format_exc())
+                write_termination_log("Exception encountered merging model checkpoints")
+                sys.exit(INTERNAL_ERROR_EXIT_CODE)
         else:
-            # copy last checkpoint into mounted output dir
-            pt_checkpoint_dir = get_highest_checkpoint(training_args.output_dir)
-            logging.info(
-                "Copying last checkpoint %s into output dir %s",
-                pt_checkpoint_dir,
-                original_output_dir,
-            )
-            shutil.copytree(
-                os.path.join(training_args.output_dir, pt_checkpoint_dir),
-                original_output_dir,
-                dirs_exist_ok=True,
-            )
+            try:
+                # copy last checkpoint into mounted output dir
+                pt_checkpoint_dir = get_highest_checkpoint(training_args.output_dir)
+                logging.info(
+                    "Copying last checkpoint %s into output dir %s",
+                    pt_checkpoint_dir,
+                    original_output_dir,
+                )
+                shutil.copytree(
+                    os.path.join(training_args.output_dir, pt_checkpoint_dir),
+                    original_output_dir,
+                    dirs_exist_ok=True,
+                )
+            except Exception as e:  # pylint: disable=broad-except
+                logging.error(traceback.format_exc())
+                write_termination_log(
+                    "Exception encountered writing output model to storage"
+                )
+                sys.exit(INTERNAL_ERROR_EXIT_CODE)
 
         # copy over any loss logs
-        train_logs_filepath = os.path.join(
-            training_args.output_dir, sft_trainer.TRAINING_LOGS_FILENAME
-        )
-        if os.path.exists(train_logs_filepath):
-            shutil.copy(train_logs_filepath, original_output_dir)
+        try:
+            train_logs_filepath = os.path.join(
+                training_args.output_dir, sft_trainer.TRAINING_LOGS_FILENAME
+            )
+            if os.path.exists(train_logs_filepath):
+                shutil.copy(train_logs_filepath, original_output_dir)
+        except Exception as e:  # pylint: disable=broad-except
```

Comment:
Since the previous behavior was to fail the job if this step fails, I changed the code to keep this behavior the same. If our users think we can claim the job as successful anyway, we can update this logic later.